packages/curve-ui-kit/src/shared/api/figma-tokens/resolve/resolver.ts

```diff
@@ -0,0 +1,74 @@
+import type { CurveFigmaDesignTokens, ScreenType } from '../tokens'
+
+export type ResolvedValue<T> = T extends { type: string; value: infer V }
```

Comment:
this file needs some documentation - what is this and why do we need to write complicated types